tools/federation/src/main/java/org/eclipse/rdf4j/federated/exception/ExceptionUtil.java

```diff
@@ -60,6 +60,10 @@ public class ExceptionUtil {
 	public static QueryEvaluationException traceExceptionSource(Endpoint endpoint, Throwable ex,
 			String additionalInfo) {
 
+		if (ex instanceof InterruptedException) {
+			Thread.currentThread().interrupt();
+		}
```

Comment:
I've tried to improve interrupt handling. Other code in RDF4J will typically re-interrupt the current thread if it needs to swallow or wrap an InterruptedException. This means that other code down the line is still able to abort their current execution, which was what the interrupt was initially signalling.